bevy_mod_scripting_core/src/systems.rs

```diff
@@ -173,21 +173,26 @@ pub fn script_event_handler<H: ScriptHost, const MAX: u32, const MIN: u32>(world
     let ctx_iter = ctxts
         .context_entities
         .iter_mut()
-        .filter_map(|(sid, (entity, o, name))| {
-            let ctx = match o {
-                Some(v) => v,
-                None => return None,
-            };
-
-            Some((
-                ScriptData {
-                    sid: *sid,
-                    entity: *entity,
-                    name,
-                },
-                ctx,
-            ))
-        });
+        .filter_map(|(entity, vec)| {
```

Comment:
what's the purpose of the sort and .flatten here?

Reply:
you're correct about the sort, that has no purpose (removed in the latest commit).
Regarding the flatten, handle_events expects a `ctxs: impl Iterator<Item = (ScriptData<'a>, &'a mut Self::ScriptContext)>`
Without flattening it, `ctx_iter` has iterators in iterators, which it does not expect.
![image](https://github.com/makspll/bevy_mod_scripting/assets/42088586/5fb21f9f-3e09-4c39-a49a-5d191669af4f)

But I'll be fully honest with you, that part was mostly written with the help of Github Copilot, so if you got a better way / preferred way of doing so, i'm open to feedback!